easybuild/framework/easyconfig/format/yeb.py

```diff
@@ -94,6 +95,9 @@ def parse(self, txt):
         """
         txt = self._inject_constants_dict(txt)
         self.parsed_yeb = yaml.load(txt)
+        # toolchain param value check, convert to dict if alt format is used
+        check_type_of_param_value('toolchain', self.parsed_yeb['toolchain'], auto_convert=True)
```

Comment:
you're not using the return value here, so if the value type was converted, you're missing it

why is this not popping up in the tests?